Source/WebDriver/glib/SessionHostGlib.cpp

```diff
@@ -344,19 +346,13 @@ void SessionHost::didStartAutomationSession(GVariant* parameters)
 
 void SessionHost::setTargetList(uint64_t connectionID, Vector<Target>&& targetList)
 {
-    // The server notifies all its clients when connection is lost by sending an empty target list.
-    // We only care about automation connection.
+    // The server notifies all its clients when connection is lost by sending an empty target list
     if (m_connectionID && m_connectionID != connectionID)
         return;
 
     ASSERT(targetList.size() <= 1);
     if (targetList.isEmpty()) {
-        m_target = Target();
-        if (m_connectionID) {
-            if (m_socketConnection)
-                m_socketConnection->close();
```

Comment:
I'm not sure I understand this change, why don't we need to call close anymore?

Reply:
Good catch. Looks like my initial version was relying on the connection being closed through other means. I added the `close()` call back, leaving the variable cleanup for the `connectionDidClose()` call.